src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1864,13 +1864,22 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
     print()
     print("Starts collecting diag info for cluster %s " % name)
 
+    # Form containerName from fqdn, as it was previously jsut the location of code is changed.
+    # https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names
+    maxContainerNameLength = 63
+    fqdn = mc.fqdn if mc.fqdn is not None else mc.private_fqdn
+    normalized_container_name = fqdn.replace('.', '-')
+    len_of_container_name = normalized_container_name.index("-hcp-")
+    if len_of_container_name == -1:
+        len_of_container_name = maxContainerNameLength
```

Comment:
Just an example, in this way, no matter what's the limitation about cluster name or fqdn, we will always get a container name no longer than 63.

```suggestion
    if len_of_container_name == -1:
        len_of_container_name = maxContainerNameLength
    else:
        len_of_container_name = min(len_of_container_name, maxContainerNameLength)
```